spec/observables/dom/ajax-spec.ts

```diff
@@ -380,6 +380,38 @@ describe('Observable.ajax', () => {
     });
   });
 
+  it('should error on timeout of asynchronous request', (done) => {
+    const obj: Rx.AjaxRequest = {
+      url: '/flibbertyJibbet',
+      responseType: 'text',
+      timeout: 10
+    };
+
+    Rx.Observable.ajax(obj)
+      .subscribe((x: any) => {
+        throw 'should not have been called';
+      }, (e) => {
+        expect(e.status).to.equal(0);
+        expect(e.xhr.method).to.equal('GET');
+        expect(e.xhr.async).to.equal(true);
+        expect(e.xhr.timeout).to.equal(10);
+        expect(e.xhr.responseType).to.equal('text');
+      });
+
+    const request = MockXMLHttpRequest.mostRecent;
+
+    expect(request.url).to.equal('/flibbertyJibbet');
+
+    setTimeout(() => {
```

Comment:
We need a way to virtualize this, perhaps with the TestScheduler, so the test doesn't take longer than a second to run. It will increase the time it takes to test everything by up to 50% if we leave this timeout in here.

Reply:
The test scheduler is available in all tests, to schedule something in "virtual time" you'd do the following:

```ts
rxTestScheduler.schedule(() => { /* thing to do */ }, 1000)

// later flush it to run all scheduled things synchronously
rxTestScheduler.flush();
```

Reply:
Great, updated to use the test scheduler instead.